template-live-examples/token-minting/contract/sources/launchpad.move

```diff
@@ -209,69 +207,67 @@ module launchpad_addr::launchpad {
             project_uri
         );
         let fa_obj = object::object_from_constructor_ref(fa_obj_constructor_ref);
-        move_to(fa_owner_obj_signer, FAOwnerObjConfig {
-            fa_obj,
-            extend_ref: object::generate_extend_ref(fa_owner_obj_constructor_ref),
-        });
-        let fa_owner_obj = object::object_from_constructor_ref(fa_owner_obj_constructor_ref);
+
         let mint_ref = fungible_asset::generate_mint_ref(fa_obj_constructor_ref);
         let burn_ref = fungible_asset::generate_burn_ref(fa_obj_constructor_ref);
         let transfer_ref = fungible_asset::generate_transfer_ref(fa_obj_constructor_ref);
-        move_to(fa_obj_signer, FAController {
-            mint_ref,
-            burn_ref,
-            transfer_ref,
-        });
-        move_to(fa_obj_signer, FAConfig {
-            mint_fee_per_smallest_unit_of_fa: *option::borrow_with_default(
-                &mint_fee_per_smallest_unit_of_fa,
-                &DEFAULT_mint_fee_per_smallest_unit_of_fa
-            ),
-            mint_limit: if (option::is_some(&mint_limit_per_addr)) {
-                option::some(MintLimit {
-                    limit: *option::borrow(&mint_limit_per_addr),
-                    mint_balance_tracker: table::new()
-                })
-            } else {
-                option::none()
-            },
-            mint_enabled: true,
-            extend_ref: object::generate_extend_ref(fa_obj_constructor_ref),
-            fa_owner_obj,
-        });
+        move_to(
+            fa_obj_signer,
+            FAController { mint_ref, burn_ref, transfer_ref }
+        );
+        move_to(
+            fa_obj_signer,
+            FAConfig {
+                mint_fee_per_smallest_unit_of_fa: *mint_fee_per_smallest_unit_of_fa.borrow_with_default(
+                    &DEFAULT_MINT_FEE_PER_SMALLEST_UNIT_OF_FA
+                ),
+                mint_limit: if (mint_limit_per_addr.is_some()) {
+                    option::some(
+                        MintLimit {
+                            limit: *mint_limit_per_addr.borrow(),
+                            mint_balance_tracker: table::new()
+                        }
+                    )
+                } else {
+                    option::none()
+                },
```

Comment:
Generally would frown upon the inline if, probably should move outside of this already big `move_to`